api/models/graphite.go

```diff
@@ -52,6 +52,42 @@ func (gr GraphiteRender) Validate(ctx *macaron.Context, errs binding.Errors) bin
 	return errs
 }
 
+type GraphiteTags struct {
+	Filter string `json:"filter"`
+	From   int64  `json:"from"`
+}
+
+type GraphiteTagsResp []GraphiteTagResp
+
+type GraphiteTagResp struct {
+	Tag string `json:"tag"`
+}
+
+type GraphiteTagDetails struct {
+	Tag    string `json:"tag"`
+	Filter string `json:"filter"`
+	From   int64  `json:"from"`
+}
+
+type GraphiteTagDetailsValueResp struct {
+	Count uint64 `json:"count"`
+	Value string `json:"value"`
+}
+
+type GraphiteTagDetailsResp struct {
+	Tag    string                        `json:"tag"`
+	Values []GraphiteTagDetailsValueResp `json:"values"`
```

Comment:
can we stick to the convention of first declaring a thing, and then the more specific thing that it relies on?
e.g. first `GraphiteTagDetailsResp`, then `GraphiteTagDetailsValueResp`
I think this is a useful convention, and we also do it in many places for functions and then the sub-functions that they call.

Reply:
:+1: done

Reply:
https://github.com/grafana/metrictank/pull/750/commits/c3cbba2576a3016092f959a5bb224fec47ca2507

Reply:
i don't have strong opinions on this and we can research this topic a bit more to see what the best practices are. see #751 but yeah for now let's stick to this approach just for consistency.